addons/account/tests/test_account_payment.py

```diff
@@ -194,6 +194,47 @@ def test_payment_move_sync_onchange(self):
             expected_liquidity_line,
         ])
 
+        # ==== Check editing the account.payment ====
+        # `partner_type` on payment is always invisible. It's supposed to be set through a context `default_` key
+        # In this case the goal of the test is to take an existing customer payment and change it to a supplier payment,
+        # which is not supposed to be possible through the web interface.
+        # So, change the payment partner_type beforehand rather than in the form view.
+        payment.action_draft()
+        payment.partner_type = 'supplier'
+        pay_form = Form(payment)
+        pay_form.currency_id = self.other_currency
+        pay_form.partner_id = self.partner_a
```

Comment:
It was already partner_a before, therefore I'm not sure why this is needed here ?